tests/js/server/aql/aql-traversal-reset-regression.js

```diff
@@ -0,0 +1,283 @@
+    {
+      _key: "408",
+      _id: ownsEdgeCollectionName + "/408",
+      _from: customerCollectionName + "/397",
+      _to: productCollectionName + "/392",
+      _rev: "_aM4ZTSG---",
+      _class: "com.arangodb.springframework.testdata.Owns"
+    },
+    {
+      _key: "410",
+      _id: ownsEdgeCollectionName + "/410",
+      _from: customerCollectionName + "/397",
+      _to: productCollectionName + "/394",
+      _rev: "_aM4ZTSG--_",
+      _class: "com.arangodb.springframework.testdata.Owns"
+    }
+  ]);
+  db[productCollectionName].insert([
+    {
+      _key: "390",
+      _id: productCollectionName + "/390",
+      _rev: "_aM4ZTPy---",
+      name: "phone",
+      _class: "com.arangodb.springframework.testdata.Product"
+    },
+    {
+      _key: "392",
+      _id: productCollectionName + "/392",
+      _rev: "_aM4ZTQG---",
+      name: "car",
+      _class: "com.arangodb.springframework.testdata.Product"
+    },
+    {
+      _key: "394",
+      _id: productCollectionName + "/394",
+      _rev: "_aM4ZTQK---",
+      name: "chair",
+      _class: "com.arangodb.springframework.testdata.Product"
+    }
+  ]);
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function traversalResetRegressionSuite() {
+  return {
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief set up
+    ////////////////////////////////////////////////////////////////////////////////
+
+    setUpAll: function() {
+      cleanup();
+      createBaseGraph();
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief tear down
+    ////////////////////////////////////////////////////////////////////////////////
+
+    tearDownAll: function() {
+      cleanup();
+    },
+
+    ////////////////////////////////////////////////////////////////////////////////
+    /// @brief test object access for path object
+    ////////////////////////////////////////////////////////////////////////////////
+    testTraversalResetCrashes: function() {
+      const query = `WITH @@product
+                       FOR e IN @@customer
+                         FILTER (FOR e1 IN 1..1 ANY e._id @@owns
+                                   FILTER e1.name == @0
+                                   RETURN 1)[0] == 1 AND
+                                (FOR e1 IN 1..1 ANY e._id @@owns
+                                   FILTER e1.name == @1
+                                   RETURN 1)[0] == 1
+                         RETURN e`;
+      const bindVars = {
+        "0": "phone",
+        "1": "phone",
+        "@product": productCollectionName,
+        "@customer": customerCollectionName,
+        "@owns": ownsEdgeCollectionName
+      };
+
+      //      var plan = db._explain(query, bindVars);
+      var actual = db._query(query, bindVars);
```

Comment:
this executes the query, fine.
but we should now check for the query results as well here!